src/lib/stores/streams/streams.store.ts

```diff
@@ -108,6 +109,36 @@ export default (() => {
     };
   }
 
+  function getIncomingTokenAmountsByUser(
+    userId: string,
+    address: string,
+  ): {
+    totalEarned: bigint;
+    amountPerSecond: bigint;
+  } {
+    const ownStreams = getStreamsForUser(userId);
+
+    if (!ownStreams) return { totalEarned: 0n, amountPerSecond: 0n };
+
+    const incomingStreamsForToken = ownStreams.incoming.filter(
+      (stream) => stream.dripsConfig.amountPerSecond.tokenAddress === address,
+    );
+
+    return incomingStreamsForToken.reduce<{ totalEarned: bigint; amountPerSecond: bigint }>(
+      (acc, stream) => {
+        const estimate = balances.getEstimateByStreamId(stream.id);
+
+        if (!estimate) throw new Error(`Unknown estimate for stream ${stream.id}`);
+
+        return {
+          totalEarned: acc.totalEarned + estimate.totalStreamed,
+          amountPerSecond: acc.amountPerSecond + estimate.currentAmountPerSecond,
+        };
+      },
+      { totalEarned: 0n, amountPerSecond: 0n },
+    );
+  }
+
```

Comment:
I think it'd be better to have this in the `balances` store to avoid tight coupling between `streams` and `balances`